src/pages/UserPage.tsx

```diff
@@ -0,0 +1,120 @@
+import { useSelector } from "react-redux";
+import { RootState } from "../redux/store";
+import {
+  Card,
+  CardContent,
+  Grid,
+  Typography,
+  CircularProgress,
+  Container,
+  Box,
+} from "@mui/material";
+import { Link } from "react-router-dom";
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import { fetchUser } from "../redux/user/user.actions";
+import { useEffect } from "react";
+import { useAppDispatch } from "../redux/hooks";
+import { Linkify } from "../utils/utilities";
+import { theme } from "../styles/theme";
+
+const defaultTheme = createTheme(theme);
+
+export const UserPage = (): JSX.Element => {
+  const dispatch = useAppDispatch();
+  const auth = useSelector((state: RootState) => state.auth);
+  const user = useSelector((state: RootState) => state.users);
+
+  useEffect(() => {
+    if (auth.loggedInUser.access_token) {
+      const userId = auth.loggedInUser.user._id;
+      if (userId) {
+        dispatch(fetchUser(userId));
+      }
+    }
+  }, [dispatch, auth]);
+
+  return (
+    <ThemeProvider theme={defaultTheme}>
+      <Container component="main" maxWidth="md" sx={{ mt: 12 }}>
+        {!user.user._id ? (
+          <CircularProgress />
+        ) : (
+          <Box
+            sx={{
+              marginTop: 8,
+              display: "flex",
+              flexDirection: "column",
+              alignItems: "center",
+            }}
+          >
+            <Grid container spacing={2}>
+              <Grid item xs={12}>
+                <Card>
+                  <CardContent>
+                    <Typography sx={{ fontWeight: "bold", fontSize: 20 }}>
+                      Username
+                    </Typography>
+                    <Typography>{user.user.username}</Typography>
+                  </CardContent>
+                </Card>
+              </Grid>
+              {user.user.posts ? (
+                <Grid item xs={12}>
+                  <Typography
+                    sx={{ fontWeight: "bold", fontSize: 20, ml: 2, mt: 2 }}
+                  >
+                    Posts
+                  </Typography>
+                  {user.user.posts.toReversed().map((post) => (
+                    <Link to={`/posts/${post._id}`} key={post._id}>
+                      <Card sx={{ my: 1 }}>
+                        <CardContent>
+                          <Typography sx={{ fontWeight: "bold", fontSize: 18 }}>
+                            {post.title}
+                          </Typography>
+                          <Linkify sx={{ my: 1 }}>{post.text}</Linkify>
+                          <Typography sx={{ fontSize: 14 }}>
+                            posted at {post.createdAt.slice(11, 16)} on{" "}
+                            {post.createdAt.slice(0, 10)}
+                          </Typography>
+                        </CardContent>
+                      </Card>
+                    </Link>
+                  ))}
+                </Grid>
+              ) : (
```

Comment:
Why don't you just do ` {user.user.posts && <Grid....}` rather than using this ternary operator to return an empty component